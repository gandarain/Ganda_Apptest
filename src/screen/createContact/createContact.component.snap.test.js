import React from 'react'
import renderer from 'react-test-renderer'

import { mockContact } from '../../fixture'

import CreateContact from './createContact.component'

jest
  .mock('../../component/header', () => 'Header')
  .mock('../../component/loader', () => 'Loader')
  .mock('react-native-image-picker', () => ({
    launchImageLibrary: jest.fn()
  }))
  .mock('react-native-config', () => ({
    BASE_URL: ''
  }))
  .mock('react-redux', () => ({
    useDispatch: jest.fn()
  }))
  .mock('@react-navigation/native', () => ({
    useNavigation: () => ({
      navigation: () => ({
        navigate: jest.fn()
      })
    })
  }))
  .mock('formik', () => ({
    Formik: 'Formik'
  }))

describe('Detail Contact Component', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  it('should render create contact', () => {
    const form = {
      photo: '',
      firstName: '',
      lastName: '',
      age: ''
    }
    const loader = false
    const setStateMock = jest.fn()

    React.useState = jest.fn().mockReturnValue([loader, {}])
    React.useState = jest.fn().mockReturnValue([form, {}])

    jest
      .spyOn(React, 'useState')
      .mockImplementationOnce(() => [form, setStateMock])
      .mockImplementationOnce(() => [loader, setStateMock])

    const rendered = renderer.create(<CreateContact />)
    const tree = rendered.toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should render edit contact', () => {
    const route = {
      params: {
        isEdit: true
      }
    }
    const form = {
      photo: mockContact.data.data.photo,
      firstName: mockContact.data.data.firstName,
      lastName: mockContact.data.data.lastName,
      age: mockContact.data.data.age.toString()
    }
    const loader = false
    const setStateMock = jest.fn()

    React.useState = jest.fn().mockReturnValue([loader, {}])
    React.useState = jest.fn().mockReturnValue([form, {}])

    jest
      .spyOn(React, 'useState')
      .mockImplementationOnce(() => [form, setStateMock])
      .mockImplementationOnce(() => [loader, setStateMock])

    const rendered = renderer.create(<CreateContact route={route} />)
    const tree = rendered.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
