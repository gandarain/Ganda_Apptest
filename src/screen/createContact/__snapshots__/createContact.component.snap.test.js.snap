// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail Contact Component should render create contact 1`] = `
<View
  accessibilityLabel="ContainerContact"
  style={
    {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <Header
    right={[]}
    showBackButton={true}
    title="Create Contact"
  />
  <Formik
    enableReinitialize={true}
    initialValues={
      {
        "age": "",
        "firstName": "",
        "lastName": "",
        "photo": "",
      }
    }
    validationSchema={
      ObjectSchema {
        "_blacklist": Set {},
        "_excludedEdges": [],
        "_mutate": undefined,
        "_nodes": [
          "age",
          "lastName",
          "firstName",
          "photo",
        ],
        "_sortErrors": [Function],
        "_typeCheck": [Function],
        "_whitelist": Set {},
        "conditions": [],
        "deps": [],
        "exclusiveTests": {},
        "fields": {
          "age": StringSchema {
            "_blacklist": Set {},
            "_typeCheck": [Function],
            "_whitelist": Set {},
            "conditions": [],
            "deps": [],
            "exclusiveTests": {
              "matches": false,
              "required": false,
            },
            "internalTests": {
              "nullable": [Function],
              "optionality": [Function],
              "typeError": [Function],
            },
            "spec": {
              "abortEarly": true,
              "coerce": true,
              "nullable": false,
              "optional": false,
              "recursive": true,
              "strict": false,
              "strip": false,
            },
            "tests": [
              [Function],
              [Function],
            ],
            "transforms": [
              [Function],
            ],
            "type": "string",
          },
          "firstName": StringSchema {
            "_blacklist": Set {},
            "_typeCheck": [Function],
            "_whitelist": Set {},
            "conditions": [],
            "deps": [],
            "exclusiveTests": {
              "matches": false,
              "required": false,
            },
            "internalTests": {
              "nullable": [Function],
              "optionality": [Function],
              "typeError": [Function],
            },
            "spec": {
              "abortEarly": true,
              "coerce": true,
              "nullable": false,
              "optional": false,
              "recursive": true,
              "strict": false,
              "strip": false,
            },
            "tests": [
              [Function],
              [Function],
            ],
            "transforms": [
              [Function],
            ],
            "type": "string",
          },
          "lastName": StringSchema {
            "_blacklist": Set {},
            "_typeCheck": [Function],
            "_whitelist": Set {},
            "conditions": [],
            "deps": [],
            "exclusiveTests": {
              "matches": false,
              "required": false,
            },
            "internalTests": {
              "nullable": [Function],
              "optionality": [Function],
              "typeError": [Function],
            },
            "spec": {
              "abortEarly": true,
              "coerce": true,
              "nullable": false,
              "optional": false,
              "recursive": true,
              "strict": false,
              "strip": false,
            },
            "tests": [
              [Function],
              [Function],
            ],
            "transforms": [
              [Function],
            ],
            "type": "string",
          },
          "photo": StringSchema {
            "_blacklist": Set {},
            "_mutate": undefined,
            "_typeCheck": [Function],
            "_whitelist": Set {},
            "conditions": [],
            "deps": [],
            "exclusiveTests": {
              "required": false,
            },
            "internalTests": {
              "nullable": [Function],
              "optionality": [Function],
              "typeError": [Function],
            },
            "spec": {
              "abortEarly": true,
              "coerce": true,
              "nullable": false,
              "optional": false,
              "recursive": true,
              "strict": false,
              "strip": false,
            },
            "tests": [
              [Function],
            ],
            "transforms": [
              [Function],
            ],
            "type": "string",
          },
        },
        "internalTests": {
          "nullable": [Function],
          "typeError": [Function],
        },
        "spec": {
          "abortEarly": true,
          "coerce": true,
          "nullable": false,
          "optional": true,
          "recursive": true,
          "strict": false,
          "strip": false,
        },
        "tests": [],
        "transforms": [],
        "type": "object",
      }
    }
  />
  <Loader
    onClose={[Function]}
    visible={false}
  />
</View>
`;

exports[`Detail Contact Component should render edit contact 1`] = `
<View
  accessibilityLabel="ContainerContact"
  style={
    {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <Header
    right={
      [
        {
          "action": [Function],
          "icon": "delete-outline",
        },
      ]
    }
    showBackButton={true}
    title="Edit Contact"
  />
  <Formik
    enableReinitialize={true}
    initialValues={
      {
        "age": "111",
        "firstName": "Bilbo",
        "lastName": "Baggins",
        "photo": "http://vignette1.wikia.nocookie.net/lotr/images/6/68/Bilbo_baggins.jpg/revision/latest?cb=20130202022550",
      }
    }
    validationSchema={
      ObjectSchema {
        "_blacklist": Set {},
        "_excludedEdges": [],
        "_mutate": undefined,
        "_nodes": [
          "age",
          "lastName",
          "firstName",
          "photo",
        ],
        "_sortErrors": [Function],
        "_typeCheck": [Function],
        "_whitelist": Set {},
        "conditions": [],
        "deps": [],
        "exclusiveTests": {},
        "fields": {
          "age": StringSchema {
            "_blacklist": Set {},
            "_typeCheck": [Function],
            "_whitelist": Set {},
            "conditions": [],
            "deps": [],
            "exclusiveTests": {
              "matches": false,
              "required": false,
            },
            "internalTests": {
              "nullable": [Function],
              "optionality": [Function],
              "typeError": [Function],
            },
            "spec": {
              "abortEarly": true,
              "coerce": true,
              "nullable": false,
              "optional": false,
              "recursive": true,
              "strict": false,
              "strip": false,
            },
            "tests": [
              [Function],
              [Function],
            ],
            "transforms": [
              [Function],
            ],
            "type": "string",
          },
          "firstName": StringSchema {
            "_blacklist": Set {},
            "_typeCheck": [Function],
            "_whitelist": Set {},
            "conditions": [],
            "deps": [],
            "exclusiveTests": {
              "matches": false,
              "required": false,
            },
            "internalTests": {
              "nullable": [Function],
              "optionality": [Function],
              "typeError": [Function],
            },
            "spec": {
              "abortEarly": true,
              "coerce": true,
              "nullable": false,
              "optional": false,
              "recursive": true,
              "strict": false,
              "strip": false,
            },
            "tests": [
              [Function],
              [Function],
            ],
            "transforms": [
              [Function],
            ],
            "type": "string",
          },
          "lastName": StringSchema {
            "_blacklist": Set {},
            "_typeCheck": [Function],
            "_whitelist": Set {},
            "conditions": [],
            "deps": [],
            "exclusiveTests": {
              "matches": false,
              "required": false,
            },
            "internalTests": {
              "nullable": [Function],
              "optionality": [Function],
              "typeError": [Function],
            },
            "spec": {
              "abortEarly": true,
              "coerce": true,
              "nullable": false,
              "optional": false,
              "recursive": true,
              "strict": false,
              "strip": false,
            },
            "tests": [
              [Function],
              [Function],
            ],
            "transforms": [
              [Function],
            ],
            "type": "string",
          },
          "photo": StringSchema {
            "_blacklist": Set {},
            "_mutate": undefined,
            "_typeCheck": [Function],
            "_whitelist": Set {},
            "conditions": [],
            "deps": [],
            "exclusiveTests": {
              "required": false,
            },
            "internalTests": {
              "nullable": [Function],
              "optionality": [Function],
              "typeError": [Function],
            },
            "spec": {
              "abortEarly": true,
              "coerce": true,
              "nullable": false,
              "optional": false,
              "recursive": true,
              "strict": false,
              "strip": false,
            },
            "tests": [
              [Function],
            ],
            "transforms": [
              [Function],
            ],
            "type": "string",
          },
        },
        "internalTests": {
          "nullable": [Function],
          "typeError": [Function],
        },
        "spec": {
          "abortEarly": true,
          "coerce": true,
          "nullable": false,
          "optional": true,
          "recursive": true,
          "strict": false,
          "strip": false,
        },
        "tests": [],
        "transforms": [],
        "type": "object",
      }
    }
  />
  <Loader
    onClose={[Function]}
    visible={false}
  />
</View>
`;
